### PR TITLE
fix: error handling for getTransactionInformation

### PR DESCRIPTION
**Pull request by yhabib**
# Motivation

`getTransactionInformation` was throwing an error while its consumers were expecting an `undefined` and they will handle the error.

# Changes

- `getTransactionInformation` should return `undefined` if the transaction type is unknown.

# Tests

- New unit tests for `mapIcpTransactionToUi` check for a toast message when an error is thrown.

# Todos

- [ ] Add entry to changelog (if necessary).
Not necessary.

### CODE REVIEW

File: frontend/src/tests/lib/utils/icp-transactions.utils.spec.ts [hunk 1]
```diff
@@ -475,6 +484,33 @@ describe("icp-transactions.utils", () => {
...
+
+      mapIcpTransactionToUi({
+        transaction,
+        accountIdentifier: from,
+        toSelfTransaction: false,
+        neuronAccounts: new Set<string>([to]),
+        swapCanisterAccounts: new Set<string>(),
+        i18n: en,
+      });
+      expect(spyToastError).toBeCalledWith({
```
Comment (mstrasinskis) [inline frontend/src/tests/lib/utils/icp-transactions.utils.spec.ts:505]: Could you please also check that it's done only once?

File: frontend/src/tests/lib/utils/icp-transactions.utils.spec.ts [hunk 2]
```diff
@@ -291,6 +292,14 @@ describe("icp-transactions.utils", () => {
   });
 
   describe("mapIcpTransactionToUi", () => {
+    let spyToastError;
+
+    beforeEach(() => {
+      spyToastError = vi
+        .spyOn(toastsStore, "toastsError")
+        .mockImplementation(vi.fn());
```
Comment (mstrasinskis) [inline frontend/src/tests/lib/utils/icp-transactions.utils.spec.ts:300]: Could you check, will it work the same way w/o `.mockImplementation(vi.fn());`? It seems like it doesn’t do anything.

File: frontend/src/lib/utils/icp-transactions.utils.ts [hunk 3]
```diff
@@ -122,9 +122,7 @@ const getTransactionInformation = (
     data = operation.Transfer;
   }
   // Edge case, a transaction will have either "Approve", "Burn", "Mint" or "Transfer" data.
-  if (data === undefined) {
-    throw new Error(`Unknown transaction type ${JSON.stringify(operation)}`);
-  }
+  if (data === undefined) return undefined;
```
Comment (mstrasinskis) [inline frontend/src/lib/utils/icp-transactions.utils.ts:125]: Could isNullish be used here?

**Review by mstrasinskis — APPROVED** [review]
Looks good, just a couple of questions/suggestions. Already approved.

File: frontend/src/tests/lib/utils/icp-transactions.utils.spec.ts (hunk 2 above)
Comment (yhabib) [reply frontend/src/tests/lib/utils/icp-transactions.utils.spec.ts:300]: It doesnt do anything but `toastsError` logs something and the test complains.  By mocking it I'm saying I don't care about the implementation as I only care about being called.

**Pull request merged**